packages/text-field/src/TextFieldPresenter.js

```diff
@@ -92,21 +96,26 @@ export default class TextFieldPresenter extends Component {
 
   static defaultProps = {
     id: generatedId(),
-    type: "text"
+    type: "text",
+    stayFocused: false
   };
 
   state = {
-    focused: false
+    focused: this.props.stayFocused
```

Comment:
Whenever state is derived from props, [`getDerivedStateFromProps`](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) should be used. This can stay `false`.

Reply:
We use the [`react-lifecycles-compat`](https://github.com/reactjs/react-lifecycles-compat) polyfill. Then `getDerivedStateFromProps` can be used regardless of the React version.

Just to clarify though, there shouldn't be a need to use that here. I was justing making a note for future reference.